File: katas/6 kyu/find-the-unique-number-1/solution.py
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def find_uniq(arr):
  arr = sorted(arr)
  for x in range(0 , len(arr)):
    if arr[x] != arr[x+1]:
      if arr[x] != arr[x-1]:
        return arr[x]
      else: return arr[x+1]
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def find_uniq(arr):
  a, b = set(arr)
  return a if arr.count(a) == 1 else b
# ----------------------------------- TEST ----------------------------------- #
@test.describe("Basic Tests")
def f():
    @test.it("Simple tests")
    def _():
        test.assert_equals(find_uniq([ 1, 1, 1, 2, 1, 1 ]), 2)
        test.assert_equals(find_uniq([ 0, 0, 0.55, 0, 0 ]), 0.55)
        test.assert_equals(find_uniq([ 3, 10, 3, 3, 3 ]), 10)

File: katas/6 kyu/find-the-unique-number-1/solution.js
// find-the-unique-number-1
/* ------------------------------- MY SOLUTION ------------------------------ */
function findUniq(arr) {
  arr.sort((a , b) => b -a)
  if (arr[2] != arr[0]) return arr[0]
  return arr[arr.length-1]
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
function findUniq(arr) {
  arr.sort((a,b)=>a-b);
  return arr[0]==arr[1]?arr.pop():arr[0]
}

function findUniq(arr) {
  return arr.find(n => arr.indexOf(n) === arr.lastIndexOf(n));
}
/* ---------------------------------- TEST ---------------------------------- */
// const chai = require("chai");
// const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
// chai.config.truncateThreshold=0;

const chai = require("chai");
const assert = chai.assert;
chai.config.truncateThreshold = 0;

describe("Example Tests", () => {
  it("Tests", () => {
    assert.strictEqual(findUniq([ 1, 0, 0 ]), 1);
    assert.strictEqual(findUniq([ 0, 1, 0 ]), 1);
    assert.strictEqual(findUniq([ 0, 0, 1 ]), 1);
    assert.strictEqual(findUniq([ 1, 1, 1, 2, 1, 1 ]), 2);
    assert.strictEqual(findUniq([ 1, 1, 2, 1, 1 ]), 2);
    assert.strictEqual(findUniq([ 3, 10, 3, 3, 3 ]), 10);
  });
});
